internal/cmd/envs.go

```diff
@@ -250,6 +250,85 @@ coder envs create my-new-powerful-env --cpu 12 --disk 100 --memory 16 --image ub
 	return cmd
 }
 
+func createEnvFromRepoCmd() *cobra.Command {
+	var (
+		org    string
+		branch string
+		name   string
+		follow bool
+	)
+
+	cmd := &cobra.Command{
+		Use:    "create-from-repo [environment_name]",
+		Short:  "create a new environment from a git repository.",
+		Args:   cobra.ExactArgs(1),
+		Long:   "Create a new Coder environment from a Git repository.",
+		Hidden: true,
+		Example: `# create a new environment from git repository template
+coder envs create-from-repo github.com/cdr/m
+coder envs create-from-repo github.com/cdr/m --branch envs-as-code`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			client, err := newClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			multiOrgMember, err := isMultiOrgMember(ctx, client, coder.Me)
+			if err != nil {
+				return err
+			}
+
+			if multiOrgMember && org == "" {
+				return xerrors.New("org is required for multi-org members")
+			}
+
+			if org == "" {
+				// Definitely janky... but since we aren't importing
+				// an image we need to get the org name somehow.
+				orgs, err := client.Organizations(ctx)
+				if err != nil {
+					return err
+				}
+				org = orgs[0].Name
```

Comment:
This get's all organizations in the deployment. We need to filter by orgs of which the user is a member. Use `lookupUserOrgs`.